modules/stochastic_tools/src/state_sim/StateProcessor.C

```diff
@@ -0,0 +1,70 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "StateProcessor.h"
+#include <stdlib.h>
+#include <string>
+#include <cmath>
+#include "MooseRandom.h"
+
+using namespace std;
+
+StateProcessor::StateProcessor(unsigned int max_time_step, int seed)
+  : _max_time_step(max_time_step),
+    _ev_times()
+    //_rand_seed(seed)
+{
+  MooseRandom::seed(seed);
+
+  //add a bunch of random time items
+  unsigned int r = MooseRandom::randl() % 10;
+  for (unsigned int i = 0; i < r; i++)
+  {
+    unsigned int addI = MooseRandom::randl() % max_time_step;
+    addEv(addI);
+  }
+}
+
+//Add the event in to the list in the proper order
+void
+StateProcessor::addEv(unsigned int time_step = 0)
+{
+  unsigned long idx = 0;
+  while ((idx < _ev_times.size()) && (_ev_times[idx] > time_step))
+    ++idx;
+
+  _ev_times.insert(_ev_times.begin() + idx, time_step);
```

Comment:
Ah! So this is why you wanted a sorted list. If your event list gets big, this insert into a vector can get pretty bad. Go here: http://www.cplusplus.com/reference/stl/ and look at the options. You have tons of options, std::set, std::list, std::priority_queue...